src/renderer/components/instance/InstanceModalContainer.ts

```diff
@@ -0,0 +1,138 @@
+import { customElement, html, LitElement, property, query, TemplateResult } from "lit-element";
+import { cache } from "lit-html/directives/cache";
+import type Instance from "../../Instance";
+import InstanceListStore from "../../store/InstanceListStore";
+import type { UnsavedDataWarning } from "./UnsavedDataWarning";
+
+const enum InstanceModalPage {
+	QuickInfo = 0,
+	Saves,
+	Servers,
+	AdvancedOptions
+}
+
+@customElement("instance-modal-container")
+export class InstanceModalContainer extends LitElement {
+	protected createRenderRoot(): this { return this; }
+
+	private instance: Instance | null = null; // reference to original instance, updated with tempInstance on save
+	@property({ type: Object }) private tempInstance: Instance | null = null; // temp clone of original instance
+	@property({ type: Number }) private currentPage: InstanceModalPage = InstanceModalPage.QuickInfo;
+
+	@query("unsaved-data-warning")
+	private unsavedDataWarning!: UnsavedDataWarning;
+	private get hasUnsavedData(): boolean {
+		return this.unsavedDataWarning.classList.contains("visible");
+	}
+
+	protected render(): TemplateResult {
+		if (this.tempInstance !== null) {
+			let content: TemplateResult;
+			switch (this.currentPage) {
+				case InstanceModalPage.QuickInfo:
+					import("./QuickInfoPage");
+					content = html`<quick-info-page .instance=${this.tempInstance} @instanceChanged=${(): void => {
+						this.requestUpdate("tempInstance");
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/manual/lukechu10/Minecraft-Box-Launcher/pullRequest?prid=6041480)